mlir/tools/mlir-rocm-runner/mlir-rocm-runner.cpp

```diff
@@ -67,17 +68,19 @@ static LogicalResult runMLIRPasses(ModuleOp m) {
   PassManager pm(m.getContext());
   applyPassManagerCLOptions(pm);
 
-  bool systemOverride = false;
-  if (tripleName.empty() && targetChip.empty() && features.empty()) {
-    systemOverride = true;
-  }
-
   int optLevel = gpuOpt.getValue();
   if (optLevel < 0 || optLevel > 3) {
     llvm::errs() << "Invalid GPU optimization level: " << optLevel << "\n";
     return failure();
   }
-  BackendUtils utils(tripleName, targetChip, features, systemOverride);
```

Comment:
You will have to keep the old behavior, or at least leave it to another review.

Without `systemOverride` you must guarantee all options passed in are valid, otherwise they can all remain to be empty.

Reply:
I just moved the empty check inside the BackendUtils constructor.